staging/src/k8s.io/kubectl/pkg/generate/versioned/service.go

```diff
@@ -210,6 +211,19 @@ func generateService(genericParams map[string]interface{}) (runtime.Object, erro
 			service.Spec.Ports[i].TargetPort = intstr.FromInt(int(port))
 		}
 	}
+	if len(params["node-port"]) > 0 {
+		nodePortStringSlice := strings.Split(params["node-port"], ",")
+		if len(service.Spec.Ports) < len(nodePortStringSlice) {
```

Comment:
I think that this should work only if the node-ports match the number of ports https://github.com/kubernetes/kubernetes/pull/98244/files#r582755570
```suggestion
		if len(service.Spec.Ports) != len(nodePortStringSlice) {
```

Reply:
Thanks, fixed.